utility/Utility.h

```diff
@@ -62,15 +62,21 @@ T join(InputIt first, InputIt last, BinaryOperation op, T empty = T{})
 }
 
 /**
-* Format the items of a map into a string as a list of key-value pairs. The map must be
-* composed of pairs of strings.
+* Format the items of a sequence container of strings into a string.
 *
+* @tparam Sequence the string sequence container (e.g. list or vector)
 * @param[in] lstr A list of strings
 * @param[in] separator The separator to use between each item
 *
 * @return the concatenated elements.
 */
-std::string asString(const std::list<std::string> &lstr, const std::string &separator = "\n");
+template <class Sequence>
+std::string asString(const Sequence &lstr, const std::string &separator = "\n")
+{
```

Comment:
static_assert(std::equal<lstr::element_type, std::string>::value, "Provided output containor container can not contain string.")